### DISCUSSION TITLE
Duck CLI: What is the correct usage of the --vault parameter, and how to download a single file from a vault?

### DISCUSSION THREAD

**Discussion by mcx808** [iterate-ch/cyberduck#15533] [Q&A] [answered]
The [documentaion](https://docs.mountainduck.io/cli/#cryptomator) is lacking any examples and doesn't properly explain the usage.

I found [this post](https://github.com/iterate-ch/cyberduck/discussions/15245) when searching for some examples which allowed me to figure out how to list a vault located in OneDrive _without_ the `--vault` parameter since it's autodetected anyway. For example to list the contents of a vault in the folder `Test` on OneDrive I can use:

`duck --list "onedrive:/My Files/Test/"`

I'm prompted for my vault password and the contents of the vault are listed.
I have also figured out auto-detection that works for upload and download the entire contents of the vault, or a pattern

`duck --upload "onedrive:/My Files/Test/" "./test.txt"`
`duck --download "onedrive:/My Files/Test/" "./"`
`alex@macOS-VM  --download "onedrive:/My Files/Test/*.txt" "./"`

But I can't figure out how to download a single file, or how the `--vault` should be used.
  
```
duck --download "onedrive:/My Files/Test/1.txt" "./"
Login successful…
Unknown. File not found.
```

`--vault` also requires a path, but supplying either `--vault "onedrive:/My Files/Test/"` or `--vault "onedrive:/My Files/Test/vault.cryptomator"` doesn't do anything. What is the correct usage of this parameter?

**Comment by dkocher** [comment, marked as answer]
The correct usage to download the file named `IMG_7010.jpeg` in the vault in `/My Files/Vault` would be 

```
--download "onedrive:/My Files/Vault/IMG_7010.jpeg" ~/Downloads/ --vault "/My Files/Vault/"
```

**Comment by dkocher** [comment]
I have opened a documentation task [^1].

[^1]: iterate-ch/docs#466
